core-bundle/tests/Routing/RouteProviderTest.php

```diff
@@ -662,9 +671,11 @@ private function mockFramework(Adapter $pageAdapter = null, Adapter $configAdapt
      */
     private function createPage(string $language, string $alias, bool $fallback = true, string $domain = '', string $scheme = null): PageModel
     {
+        mt_srand(++$this->pageModelAutoIncrement);
+
         /** @var PageModel&MockObject $page */
         $page = $this->mockClassWithProperties(PageModel::class);
-        $page->id = random_int(1, 10000);
+        $page->id = $this->pageModelAutoIncrement;
```

Comment:
This should be fine, I guess we just need different page IDs on avery `createPage` call. However, this does not really make the tests deterministic, because the page ID will be different depending on the tests run? Maybe we just need to pass an ID to the method or generate it from the arguments?

Reply:
> However, this does not really make the tests deterministic, because the page ID will be different depending on the tests run?

Why? The `pageModelAutoIncrement` member gets reset for every single test.